test/src/unit-cppapi-updates.cc

```diff
@@ -116,3 +116,47 @@ TEST_CASE(
   if (vfs.is_dir(array_name))
     vfs.remove_dir(array_name);
 }
+
+TEST_CASE(
+    "C++ API updates: empty second write", "[updates][updates-empty-write]") {
+  const std::string array_name = "updates_empty_write";
+  Context ctx;
+  VFS vfs(ctx);
+
+  if (vfs.is_dir(array_name))
+    vfs.remove_dir(array_name);
+
+  // Create
+  Domain domain(ctx);
+  domain.add_dimension(
+      Dimension::create(ctx, "d", TILEDB_STRING_ASCII, nullptr, nullptr));
+  ArraySchema schema(ctx, TILEDB_SPARSE);
+  schema.set_domain(domain).set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}});
+  Array::create(array_name, schema);
+
+  tiledb_layout_t layout = GENERATE(TILEDB_UNORDERED, TILEDB_GLOBAL_ORDER);
```

Comment:
Let's not leave TILEDB_ROW_MAJOR out of the party :blush:

Reply:
TILEDB_ROW_MAJOR requires a subarray which cannot be empty, hence impossible to do an empty write.

Reply:
right, thanks for explaining